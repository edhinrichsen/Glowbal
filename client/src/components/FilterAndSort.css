.filter-and-sort {
  text-align: left;
  position: -webkit-sticky; /* Safari */
  position: sticky;
  top: var(--unit-margin);
  padding-bottom: calc(3*var(--unit-margin));
  height: fit-content;
}
.floating-hamburger {
  border-radius: 50%;
  background-color: var(--background-color-secondary);
  height: 30px;
  width: 30px;
  margin-left: -30px; /* Center folio grid */
  z-index: 10; /* Bring to front so onClick works */
  position: sticky;
  top: var(--unit-margin);
  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.15);
  transition: box-shadow 0.3s ease;
}

.floating-hamburger .anticon {
  display: flex;
  align-content: center;
  justify-content: center;
  padding-top: 7px;
}

.floating-hamburger:hover {
  /* drop-shadow/0.12+0.8+0.5 */
  box-shadow: 0px 3px 6px -4px rgba(0, 0, 0, 0.12),
    0px 6px 16px rgba(0, 0, 0, 0.08), 0px 9px 28px 8px rgba(0, 0, 0, 0.05);
}

.anticon-menu {
  transition: color 0.3s;
}

.anticon-menu:hover {
  color: #bfbfbf;
}

.anticon-close {
  transition: color 0.3s;
}

.anticon-close:hover {
  color: #bfbfbf;
}

.ant-tooltip-inner {
  border-radius: var(--border-radius-secondary) !important;
}
