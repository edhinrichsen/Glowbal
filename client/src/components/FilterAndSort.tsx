import React, { useState } from 'react';

import { Select, Space, Input, Tooltip } from 'antd';
import { CloseOutlined, MenuOutlined } from '@ant-design/icons';

import Emoji from './Emoji';

import './FilterAndSort.css';

const { Option } = Select;

function FilterAndSort() {

    const [isOpen, setIsOpen] = useState(false);

    if (!isOpen) {
        return (
            <Tooltip title="Browse Options" placement="right" trigger={window.innerWidth <= 800 ? 'click' : 'hover'}>
                <div className="floatingHamburger" onClick={() => setIsOpen(!isOpen)}>
                    <MenuOutlined />
                </div>
            </Tooltip>
        )
    }

    return (
        <div className="container-primary floating-box">
            <Space direction="vertical" style={{ width: 'calc(100%)' }}>
                <div style={{ display: 'flex', justifyContent: 'space-between' }}>
                    <h4><strong>Filter</strong></h4>
                    <Tooltip title="Hide" placement="bottom">
                    <div className="exit-button" onClick={() => setIsOpen(!isOpen)}></div>
                </Tooltip>

                </div>

                <Select
                    mode="multiple"
                    placeholder="Project type"
                    style={{ width: '100%' }}
                >
                    <Option value="app"><Emoji symbol="ðŸ“±" /> App</Option>
                    <Option value="website"><Emoji symbol="ðŸ–¥" /> Website</Option>
                </Select>

                <Input
                    placeholder="Name"
                />

                <Select
                    mode="multiple"
                    placeholder="Tag"
                    style={{ width: '100%' }}
                >
                    <Option value="tag1">Tag 1</Option>
                    <Option value="tag2">Tag 2</Option>
                </Select>

            </Space>

            <Space direction="vertical" style={{ width: 'calc(100%)' }}>

                <br />

                <h4><strong>Sort</strong></h4>

                <Select
                    defaultValue="mostPopular"
                    style={{ width: '100%' }}
                >
                    <Option value="mostPopular">Most Popular</Option>
                    <Option value="alphabetical">Alphabetical</Option>
                </Select>

            </Space>

        </div>
    )
}

export default FilterAndSort;