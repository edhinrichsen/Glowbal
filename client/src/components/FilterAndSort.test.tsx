import React from 'react';
import { shallow } from 'enzyme';
import FilterAndSort from './FilterAndSort';

describe('Testing FilterAndSort', () => {
  it('renders without crashing', () => {
    const component = shallow(
      <FilterAndSort
        filterCallback={() => {}}
        sortCallback={() => {}}
        clearCallBack={() => {}}
        openCallBack={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
