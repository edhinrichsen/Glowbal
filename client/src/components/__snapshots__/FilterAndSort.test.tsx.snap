// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FilterAndSort renders without crashing 1`] = `
<Tooltip
  arrowPointAtCenter={false}
  autoAdjustOverflow={true}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  placement="left"
  title="Browse Options"
  transitionName="zoom-big-fast"
  trigger="hover"
>
  <div
    className="floating-hamburger"
    onClick={[Function]}
  >
    <ForwardRef(MenuOutlined) />
  </div>
</Tooltip>
`;
